channeldb/invoices.go

```diff
@@ -646,6 +652,11 @@ type InvoiceStateUpdateDesc struct {
 	// Preimage must be set to the preimage when NewState is settled.
 	Preimage *lntypes.Preimage
 
+	// HTLCPreimages set the HTLC-level preimages stored for AMP HTLCs.
+	// These are only learned when settling the invoice as a whole. Must be
+	// set when settling an invoice with non-nil SetID.
+	HTLCPreimages map[CircuitKey]lntypes.Preimage
```

Comment:
technically no, we could just reconstruct them and forgo storing them. i opted for this route since whenever we read the invoice we'd have to:
 - scan the invoice for any settled HTLCs
 - collect the setIDs of any marked settled
 - compute the HTLC set of each of those setIDs
 - reconstruct the seed/derive the preimages

It probably wouldn't be too hard to go that route, but it seemed better to compute and memoize, considering that we do a full db scan on every startup for expired invoices.